### PR TITLE
[java-client] doc updates for configuration

### PR DESCRIPTION
**Pull request by chernser**
## Summary
- configuration documentation update
- deprecation warning for client-v1

Closes https://github.com/ClickHouse/clickhouse-java/issues/1884
Closes https://github.com/ClickHouse/clickhouse-java/issues/1950
Closes  

## Checklist
- [ ] Delete items not relevant to your PR
- [ ] URL changes should add a redirect to the old URL via https://github.com/ClickHouse/clickhouse-docs/blob/main/docusaurus.config.js
- [ ] If adding a new integration page, also add an entry to the integrations list here: https://github.com/ClickHouse/clickhouse-docs/blob/main/docs/en/integrations/index.mdx

### CODE REVIEW

File: docs/en/integrations/language-clients/java/client-v1.md [hunk 1]
```diff
@@ -14,7 +14,9 @@ import CodeBlock from '@theme/CodeBlock';
 
 Java client library to communicate with a DB server thru its protocols. Current implementation supports only [HTTP interface](/docs/en/interfaces/http). The library provides own API to send requests to a server.  
 
-*Note*: this component will be deprecated soon. 
+:::warning Deprecation
+This library will be depreceted on `February, 28, 2025`. It means no releases after and support.
```
Comment (gingerwizard) [inline docs/en/integrations/language-clients/java/client-v1.md:18]: spelling `depreceted` -> `deprecated`. `It means no releases after and support. ` -> `This means no further releases and support issues will not be addressed. `

File: docs/en/integrations/language-clients/java/client-v2.md [hunk 2]
```diff
@@ -157,7 +157,7 @@ Configuration is defined during client creation. See `com.clickhouse.client.api.
...
 | setSharedOperationExecutor(ExecutorService executorService) | `executorService` - instance of executor service. | Sets executor service for operation tasks. | 
-| setClientNetworkBufferSize(int size) | - `size` - size in bytes | Sets size of a buffer in application memory space that is used to copy data back-and-forth between socket and application. Greater reduces system calls to TCP stack, but affects how much memory is spent on every connection. This buffer is also subject for GC because connections are shortlive. Also keep in mind that allocating big continious block of memory might be a problem. |
+| setClientNetworkBufferSize(int size) | - `size` - size in bytes | Sets size of a buffer in application memory space that is used to copy data back-and-forth between socket and application. Greater reduces system calls to TCP stack, but affects how much memory is spent on every connection. This buffer is also subject for GC because connections are shortlive. Also keep in mind that allocating big continious block of memory might be a problem. Default is `300,000` bytes |
```
Comment (gingerwizard) [inline docs/en/integrations/language-clients/java/client-v2.md:160]: missing full stop

File: docs/en/integrations/language-clients/java/client-v2.md [hunk 3]
```diff
@@ -363,13 +365,16 @@ Configuration options for insert operations.
   <dd>Copy buffer size. The buffer is used during write operations to copy data from user provided input stream to an output stream. Default: 8196</dd>
 
   <dt>serverSetting(String name, String value)</dt>
-  <dd>Sets individual server settings for an operation</dd>
+  <dd>Sets individual server settings for an operation.</dd>
```
Comment (gingerwizard) [inline docs/en/integrations/language-clients/java/client-v2.md:368]: Nit: can we be consistent. Ether we terminate with a full stop or we don't.

File: docs/en/integrations/language-clients/java/client-v2.md [hunk 4]
```diff
@@ -363,13 +365,16 @@ Configuration options for insert operations.
...
 
   <dt>serverSetting(String name, Collection values)</dt>
   <dd>Sets individual server settings with multiple values for an operation. Items of the collection should `String` values</dd>
 
   <dt>setDBRoles(Collection dbRoles)</dt>
   <dd>Sets DB roles to be set before executing an operation. Items of the collection should be `String` values</dd>
+
+  <dt>setOption(String option, Object value)</dt>
+  <dd>Sets a configuration option in raw format. This is not a server settings.</dd>
```
Comment (gingerwizard) [inline docs/en/integrations/language-clients/java/client-v2.md:377]: `settings` -> `setting`

File: docs/en/integrations/language-clients/java/client-v2.md [hunk 5]
```diff
@@ -547,6 +552,8 @@ Configuration options for query operations.
   <dd>Sets individual server settings with multiple values for an operation. Items of the collection should `String` values</dd>
   <dt>setDBRoles(Collection dbRoles)</dt>
   <dd>Sets DB roles to be set before executing an operation. Items of the collection should be `String` values</dd>
+  <dt>setOption(String option, Object value)</dt>
+  <dd>Sets a configuration option in raw format. This is not a server settings.</dd>
```
Comment (gingerwizard) [inline docs/en/integrations/language-clients/java/client-v2.md:556]: `settings` -> `setting`

**Review by gingerwizard — CHANGES_REQUESTED** [review]
Spelling issues but id like @mshustov to provide final approvals

**Comment by chernser** [comment]
@gingerwizard thank you for the review! Will update.

**Review by mshustov — APPROVED** [review]

**Pull request merged**